tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -291,3 +291,139 @@ def test_delete_during_rgw_svc_restart(self):
+            workload_info_list.append(resp[1])
+
+        self.log.info("Step 3: Perform Write,Read,Delete Parallely during rgw restart")
+        parallel_proc = []
+        self.log.info("Step 3a: Start Write in new process")
+        proc_write_op = multiprocessing.Process(target=self.dtm_obj.perform_write_op,
+                                                args=(bucket_name, object_prefix,
+                                                      self.test_cfg['clients'],
+                                                      self.test_cfg['samples'],
+                                                      self.test_cfg['size'],
+                                                      log_file_prefix,
+                                                      que))
+        proc_write_op.start()
+        parallel_proc.append(proc_write_op)
+        self.log.info("Step 3b: Start Read in new process")
+        # Reads, validate and delete
+        proc_read_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                               args=(workload_info_list[0], que,
+                                                     False,
+                                                     True,
+                                                     True))
+        proc_read_op.start()
+        parallel_proc.append(proc_read_op)
+        self.log.info("Step 3c: Start Deletes in new process")
+        # delete
+        proc_delete_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                                 args=(workload_info_list[1], que,
+                                                       True,
+                                                       False,
+                                                       False))
+        proc_delete_op.start()
+        parallel_proc.append(proc_delete_op)
+
+        self.log.info(
+            "Step 4: Perform Single rgw Process Restart During Write/Read/Delete Operations")
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj,
+                                            pod_prefix=const.SERVER_POD_NAME_PREFIX,
+                                            container_prefix=const.RGW_CONTAINER_NAME,
+                                            process=self.rgw_process,
+                                            check_proc_state=True)
+        assert_utils.assert_true(resp, "Failure observed during process restart/recovery")
+
+        self.log.info("Step 5: Check if all the operations were successful")
+        write_during_restart = None
+        for each in parallel_proc:
+            if each.is_alive():
+                each.join()
+            resp = que.get()
+            assert_utils.assert_true(resp[0], resp[1])
+            if isinstance(resp[1], dict):
+                write_during_restart = resp[1]
+
+        if not write_during_restart:
```

Comment:
so for read and delete , it will be taken care by background processes and no need to check anything here?

Reply:
We are checking response from all the three processes. 
as we are reading back data written during restart this additional check is added.

Reply:
You always don't need multiprocessing to achieve the parallel logic. Use multithreading and keep the test simpler.